llvm/utils/TableGen/Basic/CodeGenIntrinsics.cpp

```diff
@@ -124,6 +125,132 @@ void CodeGenIntrinsicTable::CheckTargetIndependentIntrinsics() const {
   }
 }
 
+// Return true if the given Suffix looks like mangled type. Note that this
+// check is conservative, but allows all existing LLVM intrinsic suffixes to be
+// consider as not looking like a mangling suffix.
+static bool doesSuffixLookLikeMangledType(StringRef Suffix) {
+  // Try to match against possible mangling suffixes for various types.
+  // See getMangledTypeStr() for the mangling suffixes possible. It includes
+  //  pointer       : p[0-9]+
+  //  array         : a[0-9]+[.+]
+  //  struct:       : s_/sl_[.+]
```

Comment:
The `[.+]` here is confusing, unless you really did mean "`.` or `+`". This should probably just be `.+` without the `[]`.

Reply:
I want to convey (1 or more of any character). I'll change it to `.+`.

Reply:
Done